### ISSUE TITLE
bytea columns have incorrect deserialization to raw vector

### ISSUE THREAD

**Issue by jaredk-porch** [r-dbi/RPostgres#74]
When RPostgres reads a column of `bytea` type, it deserializes the data into a raw vector by converting the _string representation_ of the data as returned by postgres to a raw vector instead of parsing the postgres hex format for bytea columns. 

As as example, the following result is incorrect:

``` R
> res.show.bug <- dbSendQuery(con, "with test as (select '\x0a'::bytea) select * from test")
> res.result <- dbFetch(res.show.bug)
> res.result
           bytea
1 5c, 78, 30, 61
> rawToChar(res.result$bytea[[1]])
[1] "\\x0a"
```

The result should clearly be (to follow POTS) equal to `as.raw(as.hexmode("0a"))`, which can be yielded from `res.result` above by applying this transformation (using `hex2raw` from the `wkb` package):

``` R
> hex2raw(rawToChar(res.result$bytea[[1]]))
[1] 0a
```

**Comment by krlmlr** [comment]
Thanks for reporting, this really should "just work".

**Comment by krlmlr** [comment]
Closed in cc596c921.
